packages/material-ui/src/Select/Select.js

```diff
@@ -192,7 +192,7 @@ Select.propTypes = {
    * This prop is required when the `native` prop is `false` (default).
    * Set to an empty string `''` if you don't want any of the available options to be selected.
    *
-   * ⚠️Value comparison to each option is a reference equality check "===", different object types will never be equal.
+   * ⚠ If the value is an object it must have reference equality with an option in order to display it.
```

Comment:
Should we add:

```suggestion
   * If the value is an object it must have reference equality with the option in order to be selected.
   * If the value is not an option, his string representation much match with the string representation of the option in order to be selected. 
```

Reply:
@oliviertassinari Thank you for you assistance!
I am trying to think of cases where non-objects `String(a) === String(b)` would not work. Maybe some long decimal values or definitely functions?  
either way it wouldnt hurt to add it into the docs

Reply:
It's meant to support 12 equal '12'.